pkg/scheduler/partition.go

```diff
@@ -342,13 +342,19 @@ func (pc *PartitionContext) AddApplication(app *objects.Application) error {
 
 	guaranteedRes := app.GetGuaranteedResource()
 	maxRes := app.GetMaxResource()
-	if !isRecoveryQueue && (guaranteedRes != nil || maxRes != nil) {
+	maxApps := app.GetMaxApps()
+	if !isRecoveryQueue && (guaranteedRes != nil || maxRes != nil || maxApps != 0) {
 		// set resources based on tags, but only if the queue is dynamic (unmanaged)
 		if queue.IsManaged() {
 			log.Log(log.SchedQueue).Warn("Trying to set resources on a queue that is not an unmanaged leaf",
 				zap.String("queueName", queue.QueuePath))
 		} else {
-			queue.SetResources(guaranteedRes, maxRes)
+			if maxApps != 0 {
+				queue.SetMaxRunningApps(maxApps)
```

Comment:
Could you cover this branch? 

We have two existing test cases: `TestAddTGApplication` and `TestAddTGAppDynamic`.

Reply:
Added the testing, thanks @pbacsko !